### PR TITLE
Sharing: Drop the usage of Jetpack.Site.prototype from SharingButtonsOptions

### PR DESCRIPTION
**Pull request by youknowriad**
follow-up to #10278
part of #8726

In this PR, I'm dropping the usage of the Jetpack.Site.prototype object from the `SharingButtonsOptions` and I rely on Redux selectors instead.
Also, I applied some of our latest coding styles to this component:
 - React Class
 - localize HoC
 - imports 

**Testing instructions**
 * Open the /sharing/buttons/$site
 * Try to edit and save the options of the sharing buttons (the second Card)
 * Try for Jetpack websites as well

### CODE REVIEW

File: client/my-sites/sharing/buttons/options.jsx [hunk 1]
```diff
@@ -95,41 +107,39 @@ module.exports = React.createClass( {
...
 		return some( this.props.buttons, { ID: 'twitter', enabled: true } );
-	},
-
-	getTwitterViaOptionElement: function() {
-		var option;
+	}
 
-		if ( ! this.isTwitterButtonEnabled() || ( this.props.site.jetpack && this.props.site.versionCompare( '3.4-dev', '<' ) ) ) {
+	getTwitterViaOptionElement() {
+		if ( ! this.isTwitterButtonEnabled() || ! this.props.isTwitterButtonAllowed ) {
```
Comment (timmyc) [inline client/my-sites/sharing/buttons/options.jsx]: Nice refactor here 👍

Comment (timmyc) [inline client/my-sites/sharing/buttons/options.jsx]: Since we have `this.props` five places in this method, do you think we should deconstruct the attributes we need into `const`s?

File: client/my-sites/sharing/buttons/buttons.jsx [hunk 2]
```diff
@@ -124,7 +124,6 @@ module.exports = protectForm( React.createClass( {
 					initialized={ this.isInitialized( [ 'settings', 'buttons' ] ) }
 					saving={ this.state.isSaving } />
 				<ButtonsOptions
-					site={ this.props.site }
 					postTypes={ this.props.postTypes.get( this.props.site.ID ) }
```
Comment (timmyc) [inline client/my-sites/sharing/buttons/buttons.jsx]: Could we grab post types from the state tree also?

File: client/my-sites/sharing/buttons/options.jsx [hunk 3]
```diff
@@ -1,92 +1,104 @@
...
+	static propTypes = {
+		buttons: PropTypes.array,
+		initialized: PropTypes.bool,
+		isJetpack: PropTypes.bool,
+		isTwitterButtonAllowed: PropTypes.bool,
+		onChange: PropTypes.func,
+		postTypes: PropTypes.array,
+		recordGoogleEvent: PropTypes.func,
+		saving: PropTypes.bool,
+		values: PropTypes.object,
```
Comment (timmyc) [inline client/my-sites/sharing/buttons/options.jsx]: missing `translate: PropTypes.func`

File: client/my-sites/sharing/buttons/buttons.jsx (hunk 2 above)
Comment (youknowriad) [inline client/my-sites/sharing/buttons/buttons.jsx]: Yes, and it's already the case in https://github.com/Automattic/wp-calypso/pull/10278 I split this refactoring into three separate PRs to avoid big ones

Comment (timmyc) [inline client/my-sites/sharing/buttons/buttons.jsx]: Yep I saw that later in the day, nice!

**Comment by timmyc** [comment]
LGTM
